src/kernels/kernelProvider.base.ts

```diff
@@ -112,12 +112,11 @@ export abstract class BaseKernelProvider implements IKernelProvider {
     protected disposeOldKernel(uri: Uri) {
         const notebook = workspace.notebookDocuments.find((nb) => nb.uri.toString() === uri.toString());
         if (notebook) {
-            traceInfoIfCI(
-                `Disposing kernel associated with ${getDisplayPath(notebook.uri)}, isClosed=${notebook.isClosed}`
-            );
-
             const kernelToDispose = this.kernelsByNotebook.get(notebook);
             if (kernelToDispose) {
+                traceInfoIfCI(
+                    `Disposing kernel associated with ${getDisplayPath(notebook.uri)}, isClosed=${notebook.isClosed}`
+                );
```

Comment:
We were getting dispose messages even when things weren't getting disposed.
The logs were too noisy and mis leading.